src/main/java/org/spongepowered/api/command/spec/CommandSpec.java

```diff
@@ -113,7 +113,21 @@ private Builder() {}
          * @return this
          */
         public Builder permission(String permission) {
-            this.permission = permission;
+            checkNotNull(permission);
+            this.permissionTester = source -> source.hasPermission(permission) ?
+                    Optional.empty() :
+                        Optional.of(Text.of("You do not have permission to use this command!"));
+            return this;
+        }
+
+        /**
+         * Set the permission tester that will be checked before using this command.
+         *
+         * @param tester The permission tester to consult
+         * @return this
+         */
+        public Builder permissionTester(Function<CommandSource, Optional<Text>> tester) {
```

Comment:
This should just accept a `Predicate` in my opinion. Add a new method for the error message, will allow it to be handled nicer as well.